src/Symfony/Bridge/Twig/Extension/CodeExtension.php

```diff
@@ -23,21 +23,19 @@
 class CodeExtension extends AbstractExtension
 {
     private $fileLinkFormat;
-    private $rootDir;
     private $charset;
     private $projectDir;
 
     /**
      * @param string|FileLinkFormatter $fileLinkFormat The format for links to source files
-     * @param string                   $rootDir        The project root directory
+     * @param string                   $projectDir     The project directory
      * @param string                   $charset        The charset
      */
-    public function __construct($fileLinkFormat, string $rootDir, string $charset, string $projectDir = null)
+    public function __construct($fileLinkFormat, string $projectDir, string $charset)
```

Comment:
argument was added in 4.2, so safe to remove technically.

So we slightly change the behavior, only in formatFile. IMHO that's fine.

Reply:
actually, we might assume the values will the same as we applied dirname() previously.